subprojects/docs/src/docs/userguide/authoringMaintainableBuildScripts.adoc

```diff
@@ -82,6 +82,19 @@ Use other collections or I/O frameworks instead of `org.gradle.util.CollectionUt
 
 Gradle plugin authors may find the Designing Gradle Plugins subsection on link:https://guides.gradle.org/designing-gradle-plugins/#restricting_the_plugin_implementation_to_gradle_s_public_api[restricting the plugin implementation to Gradle's public API] helpful.
 
+[[sec:declaring_tasks]]
+=== Declaring tasks in a build script
+
+The task API gives a build author a lot of flexibility to declare tasks in a build script.
+For optimal readability and maintainability follow these rules:
```

Comment:
An example to illustrate would really drive this home; in fact, I think this is necessary.

Reply:
done